tar/bsdtar.c

```diff
@@ -748,15 +745,14 @@ main(int argc, char **argv)
 			bsdtar->extract_flags |= ARCHIVE_EXTRACT_OWNER;
 			break;
 		case OPTION_STRIP_COMPONENTS: /* GNU tar 1.15 */
-			errno = 0;
 			tptr = NULL;
-			t = (int)strtol(bsdtar->argument, &tptr, 10);
-			if (errno || t < 0 || *(bsdtar->argument) == '\0' ||
+			l = strtol(bsdtar->argument, &tptr, 10);
+			if (l < 0 || l > 1000 || *(bsdtar->argument) == '\0' ||
```

Comment:
The 1000 here is a undocumented magic number? If we limit the strip-components depth we should probably state it in the manual page. Technically a 4096-char PATH_MAX allows us to make 2047 directories and the last directory can contain one-letter files.

Reply:
I'm happy to raise the limit here to 10 000 or even 100 000.  But there should be a limit.

Reply:
Raised limit to 100 000.

Reply:
@mmatuska Any other concerns?

Reply:
No, I am ok with a limit. What about l > 2047? But what I primarily want is to add it to the manual page that tar can strip up to 2047 components.

Reply:
I would like to have this in 3.7.4 which we should release soon due to CVE-2024-26256 (#2135)